### ISSUE TITLE
Grid is enabled for single-element collections

### ISSUE THREAD

**Issue by AlexanderSher** [Microsoft/RTVS#3090]
Test Name:	EvaluationWrapperTest.ValuesTest(script: "value.character <- 'abcdefghijklmnopqrstuvwxyz'", expectedName: "value.character", expectedValue: ""abcdefghijklmnopqrstuvwxyz"", expectedTypeName: "character", expectedClass: "character", expectedHasChildren: False, expectedCanShowDetail: False)
Test FullName:	Microsoft.VisualStudio.R.Package.Test.DataInspect.EvaluationWrapperTest.ValuesTest (3112e1e1e2da23dfbbbb28514bc91ed59399217a)
Test Source:	D:\Work\GitHub\RTVS\src\Package\Test\DataInspect\EvaluationWrapperTest.cs : line 84
Test Outcome:	Failed
Test Duration:	0:00:00.095

```cs
Result StackTrace:	
at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message) in C:\projects\fluentassertions-vf06b\Src\FluentAssertions.Net45\Execution\XUnit2TestFramework.cs:line 35
   at FluentAssertions.Execution.TestFrameworkProvider.Throw(String message) in C:\projects\fluentassertions-vf06b\Src\FluentAssertions.Net40\Execution\TestFrameworkProvider.cs:line 43
   at FluentAssertions.Execution.CollectingAssertionStrategy.ThrowIfAny(IDictionary`2 context) in C:\projects\fluentassertions-vf06b\Src\Core\Execution\CollectingAssertionStrategy.cs:line 50
   at FluentAssertions.Execution.AssertionScope.Dispose() in C:\projects\fluentassertions-vf06b\Src\Core\Execution\AssertionScope.cs:line 252
   at FluentAssertions.Equivalency.EquivalencyValidator.AssertEquality(EquivalencyValidationContext context) in C:\projects\fluentassertions-vf06b\Src\Core\Equivalency\EquivalencyValidator.cs:line 37
   at FluentAssertions.AssertionExtensions.ShouldBeEquivalentTo[T](T subject, Object expectation, Func`2 config, String because, Object[] becauseArgs) in C:\projects\fluentassertions-vf06b\Src\Shared\AssertionExtensions.cs:line 687
   at Microsoft.VisualStudio.R.Package.Test.DataInspect.VariableRHostScript.AssertEvaluationWrapper(IRSessionDataObject rdo, VariableExpectation expectation) in D:\Work\GitHub\RTVS\src\Package\Test\DataInspect\VariableRHostScript.cs:line 70
   at Microsoft.VisualStudio.R.Package.Test.DataInspect.VariableRHostScript.<EvaluateAndAssert>d__6.MoveNext() in D:\Work\GitHub\RTVS\src\Package\Test\DataInspect\VariableRHostScript.cs:line 63
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Microsoft.VisualStudio.R.Package.Test.DataInspect.EvaluationWrapperTest.<ValuesTest>d__10.MoveNext() in D:\Work\GitHub\RTVS\src\Package\Test\DataInspect\EvaluationWrapperTest.cs:line 94
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```
Result Message:	
Expected member CanShowDetail to be False, but found True.

With configuration:
- Use declared types and members
- Compare enums by value
- Try to match member by name
- Be strict about the order of items in byte arrays

**Comment by int19h** [comment]
Shouldn't be showing grids for 1-element 1D collections.

**Comment by MikhailArkhipov** [comment]
Should be probably fixed in 1.0

**Comment by int19h** [comment]
Yep, fix almost ready.
